src/Service/EndpointService.php

```diff
@@ -247,9 +238,46 @@ public function getAcceptType(): string
             return 'html';
         case 'application/vnd.openxmlformats-officedocument.wordprocessingml.document':
             return 'docx';
-            break;
+                break;
         }//end switch
 
+        return null;
+
+    }//end determineAcceptType()
+
+    /**
+     * Gets the accept type based on the request.
+     *
+     * This method breaks complexity rules but since a switch is the most efficent and performent way to do this we made a design decicion to allow it
+     *
+     * @return string The accept type
+     */
+    public function getAcceptType(): string
+    {
+        // Let's first look at the accept header.
+        $acceptHeader = $this->request->headers->get('accept');
+
+        // If the accept header does not provide useful info, check if the endpoint contains a pointer.
+        $this->logger->debug('Get Accept header');
+        if (($acceptHeader === null || $acceptHeader === '*/*') && $this->endpoint !== null && $this->endpoint->getDefaultContentType() !== null) {
+            $acceptHeader = $this->endpoint->getDefaultContentType();
+        }//end if
+
+        if (strpos($acceptHeader, ',') !== false) {
+            $acceptHeaders = explode(',', $acceptHeader);
+            foreach ($acceptHeaders as $acceptHeader) {
+                $determinedAcceptType = $this->determineAcceptType($acceptHeader);
+                if ($determinedAcceptType !== null) {
+                    return $determinedAcceptType;
+                }
+            }
+        } else {
```

Comment:
If you are using a return in the if statement. do you really need to use an else? or not?

Reply:
true